### PR TITLE
Add Documentation for a few post components

### PR DESCRIPTION
**Pull request by WenInCode**
Added some documentation for a few of the `post-` components. 
- `post-meta`
- `post-new-form`
- `post-title`
- `post-status-button`

Apparently I still haven't got this rebase thing right yet... There are a bunch of merge commits I must have pulled in.

### CODE REVIEW

File: app/components/post-meta.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
 import Ember from 'ember';
 
+/**
+  The post-meta component is a badge that presents a post's status
```
Comment (JoshSmith) [inline app/components/post-meta.js]: Something is off here. I think the component is named incorrectly. It looks like we probably want to change `post-meta` to `post-status` and move the `post.status` binding into the component itself. This PR is small enough we can tackle that here if we'd like.

File: app/components/post-new-form.js [hunk 2]
```diff
@@ -1,18 +1,50 @@
...
 
+  /**
+    @property credentials
+    @type Ember.Service
+   */
   credentials: Ember.inject.service(),
 
+  /**
+    Holds the different plcaeholder messages for each type of post
+    (task/issue/idea).
```
Comment (JoshSmith) [inline app/components/post-new-form.js]: `placeholder` is misspelled. Maybe reword to "for each post type: task, issue, idea, etc."

File: app/components/post-title.js [hunk 3]
```diff
@@ -1,15 +1,52 @@
...
 
+  /**
+    @property currentUser
+    @type Ember.Service
+   */
   currentUser: Ember.inject.service(),
 
+  /**
+    Computed alias for `currentUserIsPostAuthor` that returns whether or not the current
+    user can edit the current post.
```
Comment (JoshSmith) [inline app/components/post-title.js]: I'm not sure if we should describe what it's doing but rather what it's behavior is. "Returns whether or not the current user can edit the current post" seems simpler and better.

File: app/components/post-title.js [hunk 4]
```diff
@@ -1,15 +1,52 @@
...
+
+    @property canEdit
+    @type Boolean
+   */
   canEdit: Ember.computed.alias('currentUserIsPostAuthor'),
+
+  /**
+    Computed alias for the current user's ID.
+
```
Comment (JoshSmith) [inline app/components/post-title.js]: Same here regarding what it is.

File: app/components/post-title.js [hunk 5]
```diff
@@ -1,15 +1,52 @@
...
+    @property currentUserId
+    @type Number
+   */
   currentUserId: Ember.computed.alias('currentUser.user.id'),
+
+  /**
+    Computed alias for the post author's ID.
```
Comment (JoshSmith) [inline app/components/post-title.js]: And here.

**Comment by JoshSmith** [comment]
Very nice work here. Want to pair at some point on rebase and I can troubleshoot? Would help me learn what's at issue.

I have some minor points, so let me know what you think.

@noslouch what do you think about my points on documentation, too?

File: app/components/post-meta.js (hunk 1 above)
Comment (WenInCode) [inline app/components/post-meta.js]: That's a fair point. I think most of the components here were pretty clear what they were just by naming whereas this one took a small amount of digging. I'll look into it this evening.

**Comment by WenInCode** [comment]
Good points here - I'll update this later tonight.

File: app/components/post-meta.js (hunk 1 above)
Comment (JoshSmith) [inline app/components/post-meta.js]: Yeah, I don't know why the naming was that way anymore, but just further reinforces the need for the good work you're doing here documenting it all. Poorly named code can't survive in the light of good documentation!

**Comment by WenInCode** [comment]
@JoshSmith addressed feedback on docs -- awaiting review

File: app/components/post-status.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  ```handlebars
+  {{post-status post=post}}
+  ```
+
+  @class post-status
+  @module Component
+  @extends Ember.Component
+ */
+export default Ember.Component.extend({
+  classNames: ['post-meta'],
```
Comment (JoshSmith) [inline app/components/post-status.js]: I would rename this to match the `post-status`, which will likely mean a CSS change. Not sure if there's a `post-meta` scss file.

**Comment by JoshSmith** [comment]
@WenInCode one minor change. Fix that and rebase/merge yourself. 👍
